src/Utility/TensorFunction.h

```diff
@@ -49,6 +50,23 @@ class TensorFunc : public utl::SpatialFunction<Tensor>, public FunctionBase
   {
     return this->evaluate(X).trace();
   }
+
+  //! \brief Evaluates first derivatives of the function.
+  utl::matrix3d<Real> gradient(const Vec3& X) const
+  {
+    const size_t nsd = sqrt(this->dim());
+    utl::matrix3d<Real> result(nsd, nsd, nsd);
+    result.fill(this->evalGradient(X).data());
+
```

Comment:
remove this blank line

Reply:
ok, other projects wants blank before return so I mix things up sometimes.